Medbay Oxigen Assistant: 358/2 days (funky mode edition)

SagaStars opened the pull request:
- Añade el M.O.A (Medbay Oxigen Assistant) curara un 16 de daño de sofocación tambien estabilizando el ritmo cardiaco y dependera de una powercell. Usando un quimico especial que no puede ser obtenible de ninguna manera ( Tarda como 5 minutos en descargarse con una bateria inicial ).
- El M.O.A ya aparecera por defecto en medbay, con una bateria claro.
- Tiene sus niveles para RnD y puede ser fabricado en un protolathe

## Why It's Good For The Game
La principal razon del objeto es que servira como una alternativa a dejar a un tripulante sin pulmon 15 minutos en cryo mientras que operar a un mono humanizado para sacar un pulmon y transplantarselo, dejando al jugador poder hablar por la radio, responder pdas, ect.

En resumen un estabilizador que gaste energia en vez de quimicos de manera efectiva. Que hara el trabajo de medbay un poco mas comodo a la hora de dejar en espera a pacientes o tratar a distintos a la vez.
![image](https://user-images.githubusercontent.com/55407528/73129215-bade1080-3fb4-11ea-8792-9040f1a99759.png)

![image](https://user-images.githubusercontent.com/55407528/73128944-8f582780-3fae-11ea-85c1-8e62d087db28.png)

https://gyazo.com/8de0f18b1efcb8dbeb9c6fc279135f00
![image](https://user-images.githubusercontent.com/55407528/73128947-a1d26100-3fae-11ea-89a7-97f5955be230.png)
`tiene una sprite in_hand como todos los objetos deberian tener`


## Changelog
:cl: 
add: Nuevo equipamiento para medbay (Medbay Oxigen Assistant) aparecera en la zona de descanso de medbay
add: Nuevo medical_design para el protolathe (Medbay Oxigen Assistant)
:cl:

SagaStars committed:
esta es la tercera puta vez que subo el mismo PR

bruuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuh

SagaStars commented:
Y si evan, le puse lo de las ``cell`` 
![image](https://user-images.githubusercontent.com/55407528/69081787-2041de00-0a15-11ea-968b-c9ac26678e50.png)

Evankhell561 commented on code/HISPANIA/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -0,0 +1,11 @@
+/datum/reagent/medicine/moa_complement
+	name = "Oxygen"
```
No había ya un químico llamado oxygen?

SagaStars replied:
meme nombre time

Evankhell561 commented on code/HISPANIA/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -0,0 +1,11 @@
+
+/datum/reagent/medicine/moa_complement/on_mob_life(mob/living/M)
+		var/update_flags = STATUS_UPDATE_NONE
+		update_flags |= M.adjustOxyLoss(-16, FALSE)
```
todos los otros químicos que usan `update_flags |= algo llevan `al final `return ..() | update_flags`. No sé para qué es pero seguro es importante. 

![image](https://user-images.githubusercontent.com/49106866/69382576-ca2b9f80-0c8d-11ea-950f-576ab14139c5.png)

SagaStars replied:
hmm... ya lo pongo

Evankhell561 commented on code/HISPANIA/modules/reagents/reagents_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+	var/status = 3
+	var/hitcost = 64
+	var/obj/item/stock_parts/cell/cell = null
+	var/mode = IV_INJECT
```
No vi que llegaste a usar esta variable nunca, ¿para qué o dónde la usas?

SagaStars replied:
Fuck

Evankhell561 commented on code/HISPANIA/modules/reagents/reagents_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+#define IV_DRAW 0
+#define IV_INJECT 1
```
Veo que has definido esto para usarlos en la variable "mode" pero no vi que la hayas usado para otra cosa que no sea definirla. Si no vas a usar "mode" en ningun lado estas dos definiciones no son necesarias.

Evankhell561 commented on code/HISPANIA/modules/reagents/chemistry/reagents/medicine.dm:
```diff
@@ -0,0 +1,11 @@
+	reagent_state = LIQUID
+	heart_rate_decrease = 1
+	can_synth = FALSE
+	taste_description = "a safe refuge"
```
Tiene el mismo `taste_description ` que la cryoxadona. No creo que dos bebidas debieran compartir la misma `taste_description `

SagaStars replied:
vaya caprichos evan

SagaStars replied:
ya lo pongo po

Evankhell561 commented on code/HISPANIA/modules/reagents/reagents_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+
+/obj/item/reagent_containers/moa/process()
+	if(!injection_target)
+		end_processing()
```
Me gusta como terminas el proceso y lo inicas solo cuando es necesario, me robaré este código a futuro :3

SagaStars replied:
**epico**

Evankhell561 commented on code/HISPANIA/modules/reagents/reagents_containers/moa.dm:
```diff
@@ -0,0 +1,172 @@
+	end_processing()
+	return ..()
+
+/obj/item/reagent_containers/moa/pickup(mob/user)
```
me robo esto de actualizar iconos también~

SagaStars committed:
Weas de evan

SagaStars committed:
Merge branch 'master' into moaepico

SagaStars committed:
Merge branch 'master' into moaepico

SagaStars committed:
REMASTERED EDITION HD FULL

- Sonidos y sprites nuevas.
- Tiene niveles para RnD.
- Puede ser construido en un protolathe.
-SPRITES INHAAND

SagaStars committed:
Lo quita del mapa y lo pone en un locker

SagaStars committed:
Merge branch 'master' into moaepico

SagaStars committed:
the fucking end

SagaStars committed:
Merge branch 'master' into moaepico

SagaStars committed:
Merge branch 'master' into moaepico

Evankhell561 commented on code/HISPANIA/modules/reagents/reagents_containers/moa.dm:
```diff
@@ -0,0 +1,181 @@
+	w_class = WEIGHT_CLASS_HUGE
+	amount_per_transfer_from_this = 1
+	var/status = 3
+	var/hitcost = 64
```
pon esto en 85.
